crates/atuin-desktop-runtime/src/blocks/sub_runbook.rs

```diff
@@ -0,0 +1,1206 @@
+        Block::SubRunbook(self)
+    }
+
+    fn create_state(&self) -> Option<Box<dyn BlockState>> {
+        Some(Box::new(SubRunbookState::default()))
+    }
+
+    async fn execute(
+        self,
+        context: ExecutionContext,
+    ) -> Result<Option<ExecutionHandle>, Box<dyn std::error::Error + Send + Sync>> {
+        tracing::trace!("Executing sub-runbook block {id}", id = self.id);
+
+        // Check if runbook reference is specified
+        if self.runbook_ref.is_empty() {
+            let _ = context.block_started().await;
+            let _ = context
+                .update_block_state::<SubRunbookState, _>(self.id, |state| {
+                    state.status = SubRunbookStatus::Failed {
+                        error: "No runbook selected".to_string(),
+                    };
+                })
+                .await;
+            let _ = context
+                .block_failed("No runbook selected".to_string())
+                .await;
+            return Ok(Some(context.handle()));
+        }
+
+        // Check if runbook loader is available
+        let runbook_loader = match context.runbook_loader() {
+            Some(loader) => loader.clone(),
+            None => {
+                let _ = context.block_started().await;
+                let _ = context
+                    .update_block_state::<SubRunbookState, _>(self.id, |state| {
+                        state.status = SubRunbookStatus::Failed {
+                            error: "Sub-runbook execution not available".to_string(),
+                        };
+                    })
+                    .await;
+                let _ = context
+                    .block_failed(
+                        "Sub-runbook execution not available (no runbook loader configured)"
+                            .to_string(),
+                    )
+                    .await;
+                return Ok(Some(context.handle()));
+            }
+        };
+
+        // Clone context to get handle for return value (original moves into spawned task)
+        let handle_context = context.clone();
+        let block_id = self.id;
+        let runbook_ref = self.runbook_ref.clone();
+        let export_env = self.export_env;
+        // Use runbook_name if set, otherwise fall back to display_id
+        let runbook_name = self
+            .runbook_name
+            .clone()
+            .unwrap_or_else(|| self.runbook_ref.display_id());
+
+        tokio::spawn(async move {
+            // Mark block as started
+            let _ = context.block_started().await;
+
+            // Update state to loading
+            let _ = context
+                .update_block_state::<SubRunbookState, _>(block_id, |state| {
+                    state.status = SubRunbookStatus::Loading;
+                })
+                .await;
+
+            // Check for recursion before loading (use display_id for stack tracking)
+            let stack_id = runbook_ref.display_id();
```

Comment:
Is there any risk of two different runbooks trying to load the same runbook but with a different display_id? e.g. `user/runbook` vs `user/runbook:latest`? Maybe could check again just before execution after all the UUIDs are resolved if so